webapp/packages/core-blocks/src/FormControls/InputAutocompletionMenu.tsx

```diff
@@ -0,0 +1,156 @@
+
+interface AutocompletionProps {
+  proposals: InputAutocompleteProposal[];
+  inputRef: React.RefObject<HTMLInputElement | HTMLTextAreaElement>;
+  className?: string;
+  onSelect?: (proposal: InputAutocompleteProposal) => void;
+}
+
+const CONTEXT_INPUT_OFFSET_Y = 3;
+const DEBOUNCE_DELAY = 300;
+
+export const InputAutocompletionMenu = observer(function InputAutocompletionMenu({ className, proposals, inputRef, onSelect }: AutocompletionProps) {
+  const styles = useS(style, BaseDropdownStyles);
+  const menuRef = useRef<IMenuState>();
+  const state = useObservableRef(
+    () => ({
+      x: 0,
+      y: 0,
+      inputValue: '',
+    }),
+    {
+      x: observable.ref,
+      y: observable.ref,
+      inputValue: observable.ref,
+    },
+    false,
+  );
+  const contextMenuPosition = getComputed(() => ({
+    position: { x: state.x, y: state.y },
+    handleContextMenuOpen: () => {},
+  }));
+
+  function handleSelect(proposal: InputAutocompleteProposal) {
+    menuRef.current?.hide();
+    onSelect?.(proposal);
+  }
+
+  function handleKeyDown(event: any) {
+    switch (event.key) {
+      case 'Escape':
+        menuRef.current?.hide();
+        break;
+      case 'ArrowDown':
+      case 'ArrowUp':
+        menuRef.current?.first();
+        break;
+      default:
+        break;
+    }
+  }
+
+  const handleInput = useMemo(
+    () =>
+      debounce((event: any) => {
+        state.inputValue = event.target.value;
+      }, DEBOUNCE_DELAY),
+    [state],
+  );
+
+  useLayoutEffect(() => {
+    if (!inputRef.current) {
+      return;
+    }
+
+    const inputElement = inputRef.current;
+    const caretPosition = inputElement.selectionStart || 0;
+    const inputStyle = window.getComputedStyle(inputElement);
+
+    const span = document.createElement('span');
+    span.style.position = 'absolute';
+    span.style.visibility = 'hidden';
+    span.style.whiteSpace = 'pre';
+    span.style.fontFamily = inputStyle.fontFamily;
+    span.style.fontSize = inputStyle.fontSize;
+    span.style.fontWeight = inputStyle.fontWeight;
+    span.style.letterSpacing = inputStyle.letterSpacing;
+    span.style.lineHeight = inputStyle.lineHeight;
+    span.textContent = inputElement.value.slice(0, caretPosition);
+
+    document.body.appendChild(span);
+    const spanRect = span.getBoundingClientRect();
+    const letterWidth = spanRect.width / span.textContent.length;
+    document.body.removeChild(span);
+
+    state.x = spanRect.width + letterWidth;
+    state.y = spanRect.height + CONTEXT_INPUT_OFFSET_Y;
+  }, [state.inputValue, inputRef.current]);
+
+  useEffect(() => {
+    inputRef.current?.addEventListener('keydown', handleKeyDown);
+    inputRef.current?.addEventListener('input', handleInput);
+
+    return () => {
+      inputRef.current?.removeEventListener('keydown', handleKeyDown);
+      inputRef.current?.removeEventListener('input', handleInput);
+    };
+  }, [inputRef.current]);
```

Comment:
`React.RefObject` can't be used this way because:
1. it's passed as a component property
2. it's mutated by react and will not cause child component (this component) to re-render when the `current` changed

if you need some logic related to some input ref please place this logic in some hook and use this hook in the same component that passes ref to input to avoid synchronization bugs